testsuite/smoke-test/src/state_sync.rs

```diff
@@ -950,6 +1003,38 @@ async fn stop_validator_and_delete_storage(swarm: &mut LocalSwarm, validator: Ac
     validator.clear_storage().await.unwrap();
 }
 
+/// Verifies that the oldest ledger info, pruning metrics and first
+/// ledger info are all correctly aligned after a fast sync.
+async fn verify_fast_sync_version_and_metrics(node: &mut LocalNode, sync_to_genesis: bool) {
+    // Verify the oldest ledger info for the node
+    verify_oldest_version_after_fast_sync(node.rest_client(), sync_to_genesis).await;
+
+    // Verify the node's pruning metrics
+    let inspection_client = node.inspection_client();
+    verify_pruning_metrics_after_fast_sync(inspection_client, sync_to_genesis).await;
+
+    // Verify that the ledger info exists at version 0
+    verify_first_ledger_info(node);
+}
+
+/// Verifies that the ledger info at version 0 exists in the given node's DB
+fn verify_first_ledger_info(node: &mut LocalNode) {
+    // Get the DB path for the node
+    let db_path = node.config().base.data_dir.as_path();
+    let mut db_path_buf = db_path.to_path_buf();
+    db_path_buf.push("db");
+
+    // Stop the node to prevent any DB contention
+    node.stop();
+
+    // Verify that the ledger info exists at version 0
+    let aptos_db = AptosDB::new_for_test(db_path_buf.as_path());
+    aptos_db.get_epoch_ending_ledger_info(0).unwrap();
```

Comment:
nit: use _expect_ instead of unwrap to explain why this should fail?

Reply:
Probably a personal taste, but I tend to avoid `expect` in tests, unless the `unwrap` is not obvious. The problem I find with `expect()` in tests is that folks write things like `expect("This should exist!");` and that doesn't really help 😄